x-pack/plugins/monitoring/server/lib/cluster/flag_supported_clusters.ts

```diff
@@ -41,7 +50,7 @@ async function findSupportedBasicLicenseCluster(
               bool: {
                 should: [
                   { term: { type: 'kibana_stats' } },
-                  { term: { 'metricset.name': 'stats' } },
```

Comment:
@klacabane mentioned in https://github.com/elastic/kibana/issues/120825 that the the documents coming from metricbeat standalone mode don't contain a `data_stream.dataset`. We might need to keep these terms in to ensure those documents get matched as well.

Reply:
@klacabane mentioned in https://github.com/elastic/kibana/issues/120825 that the the documents coming from metricbeat standalone mode don't contain a `data_stream.dataset`. We might need to keep these terms in to ensure those documents get matched as well (even though they're not in metricbeat-* indices).

Reply:
At the moment it seems that `metricset.name` is the only choice to match the metricbeat documents so we should keep it, but given that metricbeat 8 will write to data streams, maybe we could have the property on these documents as well ?

Reply:
I understand it wouldn't currently match `data_stream.dataset`, but I was under the impression it would match `{ term: { type: 'kibana_stats' } }`.  On 7.16 in cloud which uses the -mb indices, it has this field in the source and works with the queries:
![Screen Shot 2021-12-16 at 9 54 06 AM](https://user-images.githubusercontent.com/1676003/146394886-a87f371a-bbb9-4cbb-b8da-c2fbaee35604.png)
It has both `type` and `metricset.name`.  Is this going to change in 8.0?

Reply:
I understand it wouldn't currently match `data_stream.dataset`, but I was under the impression it would match `{ term: { type: 'kibana_stats' } }` for example.  On 7.16 in cloud which uses the -mb indices, it has this field in the source and works with the queries:
![Screen Shot 2021-12-16 at 10 12 39 AM](https://user-images.githubusercontent.com/1676003/146397485-d7470a1b-034a-4656-ad7e-bd35a34f68ba.png)

It has both `type` and `metricset.name`.  I assume this wont change in 8.0?

Reply:
I believe the `type` property is part of the `xpack.enabled: true` logic that creates documents with the legacy schema. Since this setting won't be supported anymore in metricbeat modules starting 8.0 it'll always create documents with the new format which only includes `metricset.name`. We could maybe set an alias but the naming are sometimes different

Reply:
Thanks @klacabane .  Okay if that is the case, then it sounds like we would need `metricset.name`.  But if, as you mentioned, metricbeat 8 will write to data streams, then we would not need it?

Reply:
That would be great to have a common interface to filter these documents but it looks like the `data_stream` property is part of the agent integrations and doesn't appear in metricbeat standalone. @sayden do you have context on this ? Could we add the `data_stream` property to documents indexed by metricbeat standalone ?

Reply:
Jason did a good job when he did the matrix for "situational awareness". In general all this creates some confusion because we must think in multi-version scenarios. Generally speaking this is how metricbeat is working now:

* metricbeat 8.x, index in `.monitoring` and legacy schema, stuff like `metricset.name` 
* metricbeat 7.x  index in `.monitoring` and legacy schema, stuff like `metricset.name` 
* agent (only 8.x) index into data streams and some things are similar but there are not metricbeat or metricset fields, instead you have `data_stream` with similar information or ECS fields https://www.elastic.co/guide/en/ecs/current/index.html

So, for example, answering @klacabane question:

> it looks like the data_stream property is part of the agent integrations and doesn't appear in metricbeat standalone

Correct, since its inception Agent only works with data streams and has no connection to Metricbeat (it has, but it's better to think that it doesn't).

> Could we add the data_stream property to documents indexed by metricbeat standalone ?

No, I'm afraid we can't. `data_stream` objects are solely for Agent and data indexed in elasticsearch data_streams. Anything already included in `data_stream.*` has some equivalent place in Metricbeat though.

> I believe the type property is part of the xpack.enabled: true

Mostly true but to extend the comment a bit more, the `type` property is part of the legacy monitoring (legacy monitoring is the one that elasticsearch was using to self monitor itself, then Metricbeat inherit that schema years later as technical debt to avoid changes in Kibana). Each `type` had a 1:1 relationship with a Metricbeat metricset (which in turn, had a 1:1 relationship with a Agent data stream). 

For example `type: "cluster_stats"` is a legacy field, originally indexed in `.monitoring-*` indices. In Metricbeat 8.x the equivalent field would be `metricset.name: "cluster_stats"` (many times with the same value, but not always). In Agent it will be `data_stream.dataset`. I said that they don't always have the same value: for example `type: ccr_stats` has now a value of `metricset.name: ccr` and all the data will be found in `elasticsearch.ccr`.

Following with the `cluster_stats` example, in legacy and metricbeat 7.x, all data would be found in `cluster_stats.*`. In metricbeat and agent it will be in `elasticsearch.cluster_stats.*`.

I think the main difficulty in Kibana is to switch between one and the other. A Kibana 8.x using Metricbeat 8.x will need to look for `metricset.name` to find that old `type` field (in the example, to find `cluster_stats`). While a Kibana 8.x using Agent must look into `data_stream.dataset`. Kibana 8.x using Metricbeat 7.8, for example, have `type` fields, but I think Jason said at some point that we accept that we won't support such scenarios anymore.

Reply:
Thanks @sayden.  Sounds like we should keep `metricset.name` as metricbeat may not continue to send `type`.  That we should think in terms of legacy using `type`, metricbeat using `metricset.name` and agent using `data_stream.dataset` and have all three in the `should`.

> agent (only 8.x) index into data streams and some things are similar but there are not metricbeat or metricset fields, instead you have data_stream with similar information or ECS fields https://www.elastic.co/guide/en/ecs/current/index.html

Is this a recent change?   I was running 8.1.0 agent snapshot recently and I did notice that docs indexed by the agent had `metricset.name`.  You can see them in these examples docs: https://github.com/elastic/integrations/tree/master/packages/elasticsearch